pkg/config/config.go

```diff
@@ -57,6 +59,7 @@ type DetailedInfo struct {
 	Subnet           *ec2.Subnet
 	InstanceTypeInfo *ec2.InstanceTypeInfo
 	SecurityGroups   []*ec2.SecurityGroup
+	TagSpecs         []*ec2.TagSpecification
```

Comment:
Why isn't `ec2.UserData` type part of this struct? Just checking if DetailedInfo is used in multiple places.

Reply:
My understanding of _DetailedInfo_ is it houses the "formal" structures required for interacting with aws sdk. And if work needs to be done to the config provided by the user (fronted via _SimpleInfo_) to convert into these formal structures, then it should be part of _DetailedInfo_.

Because UserData does not require such additional work I did not place it in _DetailedInfo_. Compare this to `TagSpecification` which requires work to convert the user's input into the necessary aws struct. This is also why I didn't put `IamInstanceProfile` in   _DetailedInfo_.

Ex:  `input.UserData = aws.String(base64.StdEncoding.EncodeToString(userDataRaw))`